vector-im/element-x-android
# Handle 'invalid server' error in server selection screen properly.

## Changes

Adds a `ChangeServerError` that maps any throwable into something we can present in the `ChangeServerView`.

Addresses #210 .

## Tests

1. Go to the server selection screen.
2. Enter an invalid URL, and submit the form. An error message should replace the default footer message.
3. Modify the URL or tap on the 'clear' button in the textfield. The default info message should be restored.
4. Input a real homeserver with no sliding sync support. A dialog with a custom action should be displayed instead of the footer error.
